### PR TITLE
Add voice assistant announce

### PR DESCRIPTION
**Pull request by synesthesiam**
# What does this implement/fix?

Adds a new `VoiceAssistantAnnounce` message that will allow media URLs to be announced if the device supports it.

* https://github.com/esphome/aioesphomeapi/pull/939
* https://github.com/home-assistant/core/pull/125157
* https://github.com/esphome/voice-kit/pull/64

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Comment by synesthesiam** [comment]
Leaving as a draft since this is still under discussion.

File: esphome/components/api/api.proto [hunk 1]
```diff
@@ -1554,6 +1554,14 @@ message VoiceAssistantTimerEventResponse {
   bool is_active = 6;
 }
 
+message VoiceAssistantAnnounce {
```
Comment (jesserockz) [inline esphome/components/api/api.proto]:
```suggestion
message VoiceAssistantAnnounceRequest {
```

File: esphome/components/api/api.proto [hunk 2]
```diff
@@ -1554,6 +1554,29 @@ message VoiceAssistantTimerEventResponse {
...
+
+  string media_id = 1;
+}
+
+message VoiceAssistantTextToSpeechFormat {
+  string format = 1;
+  uint32 sample_rate = 2;
+  uint32 num_channels = 3;
+}
+
```
Comment (jesserockz) [inline esphome/components/api/api.proto]: I can see this is not being used yet, It should be removed from this PR, or added to the `aioesphomeapi` one to match please.

File: esphome/components/api/api.proto [hunk 3]
```diff
@@ -1554,6 +1554,29 @@ message VoiceAssistantTimerEventResponse {
...
+  string text = 2;
+}
+
+message VoiceAssistantAnnounceFinished {
+  option (id) = 121;
+  option (source) = SOURCE_SERVER;
+  option (ifdef) = "USE_VOICE_ASSISTANT";
+
+  string media_id = 1;
+}
```
Comment (bdraco) [inline esphome/components/api/api.proto]: We have a conflict with allocation of `id`s.  119 is the next available one, or we are missing a merge/change on the aioesphomeapi side

Comment (synesthesiam) [reply esphome/components/api/api.proto]: 119 was the MediaPlayerSupported format, but I put an id on it not realizing that it wasn't needed (it's a sub-message). Should we use 119 and 120 here instead?

File: esphome/components/api/api.proto (hunk 2 above)
Comment (synesthesiam) [reply esphome/components/api/api.proto]: Removed

File: esphome/components/api/api.proto (hunk 3 above)
Comment (jesserockz) [reply esphome/components/api/api.proto]: 119 was accidentally assigned to `MediaPlayerSupportedFormat` but it doesnt need an id as it is not a message that is sent or received by itself, I will fix that to remove the id there. Then yes this PR needs to change to 119 as the first id.

Comment (jesserockz) [reply esphome/components/api/api.proto]: https://github.com/esphome/esphome/pull/7406

Comment (jesserockz) [reply esphome/components/api/api.proto]: https://github.com/esphome/aioesphomeapi/pull/948

**Review by jesserockz — APPROVED** [review]

**Pull request merged**